File: src/graphs/cpp/prim/test/test_prim.cpp
#include "prim.h"
#include "gtest/gtest.h"

TEST(TestPrim, nonEmptyGraph) {
  Graph<int> graph;

  graph.addEdge(Vertex<int>(0), Vertex<int>(8), 1);
  graph.addEdge(Vertex<int>(0), Vertex<int>(3), 10);
  graph.addEdge(Vertex<int>(1), Vertex<int>(2), 1);
  graph.addEdge(Vertex<int>(1), Vertex<int>(7), 6);
  graph.addEdge(Vertex<int>(1), Vertex<int>(9), 8);
  graph.addEdge(Vertex<int>(1), Vertex<int>(3), 11);
  graph.addEdge(Vertex<int>(1), Vertex<int>(5), 8);
  graph.addEdge(Vertex<int>(1), Vertex<int>(4), 11);
  graph.addEdge(Vertex<int>(2), Vertex<int>(3), 7);
  graph.addEdge(Vertex<int>(2), Vertex<int>(8), 2);
  graph.addEdge(Vertex<int>(2), Vertex<int>(1), 4);
  graph.addEdge(Vertex<int>(2), Vertex<int>(7), 6);
  graph.addEdge(Vertex<int>(2), Vertex<int>(6), 4);
  graph.addEdge(Vertex<int>(2), Vertex<int>(5), 3);
  graph.addEdge(Vertex<int>(3), Vertex<int>(5), 11);
  graph.addEdge(Vertex<int>(4), Vertex<int>(3), 12);
  graph.addEdge(Vertex<int>(5), Vertex<int>(8), 2);
  graph.addEdge(Vertex<int>(6), Vertex<int>(1), 1);
  graph.addEdge(Vertex<int>(6), Vertex<int>(2), 14);
  graph.addEdge(Vertex<int>(6), Vertex<int>(3), 10);
  graph.addEdge(Vertex<int>(7), Vertex<int>(3), 5);
  graph.addEdge(Vertex<int>(7), Vertex<int>(4), 1);
  graph.addEdge(Vertex<int>(8), Vertex<int>(5), 6);
  graph.addEdge(Vertex<int>(8), Vertex<int>(6), 7);

  EXPECT_EQ(prim(graph), 27);
}

File: src/graphs/python/prim/tests/test_prim.py
import unittest

import os
import sys

file_dir = os.path.dirname(os.path.dirname(__file__))
sys.path.append(file_dir + "/src")

from graph import Vertex, Edge, Graph
from prim import prim


class TestPrim(unittest.TestCase):
    def test_prim_str_vertices(self):

        graph = Graph()

        graph.add_edge(Edge(Vertex("A"), Vertex("D"), 2))
        graph.add_edge(Edge(Vertex("A"), Vertex("G"), 3))
        graph.add_edge(Edge(Vertex("A"), Vertex("B"), 1))
        graph.add_edge(Edge(Vertex("B"), Vertex("E"), 6))
        graph.add_edge(Edge(Vertex("B"), Vertex("F"), 7))
        graph.add_edge(Edge(Vertex("F"), Vertex("D"), 10))
        graph.add_edge(Edge(Vertex("F"), Vertex("C"), 12))
        graph.add_edge(Edge(Vertex("E"), Vertex("G"), 9))

        self.assertEqual(prim(graph), 31)

    def test_prim_int_vertices(self):

        graph = Graph()

        graph.add_edge(Edge(Vertex(0), Vertex(1), 1))
        graph.add_edge(Edge(Vertex(0), Vertex(4), 1))
        graph.add_edge(Edge(Vertex(1), Vertex(0), 1))
        graph.add_edge(Edge(Vertex(1), Vertex(3), 1))
        graph.add_edge(Edge(Vertex(1), Vertex(4), 1))
        graph.add_edge(Edge(Vertex(1), Vertex(2), 1))
        graph.add_edge(Edge(Vertex(2), Vertex(3), 1))
        graph.add_edge(Edge(Vertex(2), Vertex(1), 1))
        graph.add_edge(Edge(Vertex(3), Vertex(1), 1))
        graph.add_edge(Edge(Vertex(3), Vertex(2), 1))
        graph.add_edge(Edge(Vertex(3), Vertex(4), 1))

        self.assertEqual(prim(graph), 4)


if __name__ == "__main__":
    unittest.main()
